src/Microsoft.AspNetCore.Hosting/Internal/RequestServicesContainerMiddleware.cs

```diff
@@ -13,6 +14,8 @@ public class RequestServicesContainerMiddleware
         private readonly RequestDelegate _next;
         private readonly IServiceProvider _services;
 
+        private IServiceScopeFactory _scopeFactory;
+
         public RequestServicesContainerMiddleware(RequestDelegate next, IServiceProvider services)
```

Comment:
Don't want to pay the cost. We'd have to measure it to see how it affects scenarios where you didn't ask for any services. If it was negligible then sure.